src/components/workAllocation/WorkAllocation.tsx

```diff
@@ -382,25 +382,28 @@ export default function WorkAllocation() {
                     type={'number'}
                     label="DNAP study ID"
                     name="ssStudyId"
+                    value={values.ssStudyId}
                     onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
                       const ssId = Number(e.currentTarget.value);
                       stanCore
                         .GetDnapStudy({ ssId: ssId })
                         .then((study) => {
                           if (study && study.dnapStudy) {
                             setFieldValue('studyName', study.dnapStudy.name);
-                            setFieldValue('ssStudyId', ssId);
+                          } else {
+                            setFieldValue('studyName', 'undefined');
                           }
                         })
                         .catch((e) => {
                           setFieldValue('studyName', 'undefined');
```

Comment:
Why we need to set this to 'undefined' string?. Would it be better to make it to an empty string?

Reply:
it seems like when setting to an empty string, when validate to read it fails  (ie: values.studyName return false )

Reply:
Where are we validating this?

Reply:
I think in the default form value as well it can be initialised to '' instead of undefined

Reply:
The validation is done just below L403, to display the study name of an error message

Reply:
I think it shouldn't be a problem because TS validates an empty string as false as well when used with && or | operator -  “falsy” values (e.g. the empty string, 0, NaN, and, well, false)
(https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html)

Reply:
that's exactly the issue, as I only need to display the error message when the user enters invalid study id